### ISSUE TITLE
Inconsistency of // for reals and complexes

### ISSUE THREAD

**Issue by dpsanders** [JuliaLang/julia#16282]
```
julia> 3 // 4.5
ERROR: MethodError: `//` has no method matching //(::Int64, ::Float64)
Closest candidates are:
  //(::Integer, ::Integer)
  //(::Integer, ::Rational{T<:Integer})
  //(::Complex{T<:Real}, ::Real)
  ...
```
but 
```
julia> 3 // 4.5im
0//1 - 2//3*im

julia> 3 // 4.7im
ERROR: OverflowError()
 in * at rational.jl:188
 in // at rational.jl:46
 in // at rational.jl:42
```
It seems to me that `3 // 4.5` should give the result of `3 / 4.5`, i.e. `Float64`.
The fact that the complex `//` tries to convert the float to a rational, and succeeds only if the float is exactly representable, seems to be a very bad idea.

**Comment by dpsanders** [comment]
Cc @simonbyrne

**Comment by dpsanders** [comment]
The alternative is that any // should give an error with any kind of AbstractFloat.

**Comment by tkelman** [comment]
fixed by #17206
